CONTRIBUTING.md

```diff
@@ -7,100 +7,198 @@
 to the style guidelines. Be aware this formatting method might need user
-supervision.
+supervision. To have `clang-format` ignore certain special cases, use the
+following special comments:
+
+```c
+int formatted_code;
+// clang-format off
+    void    unformatted_code  ;
+// clang-format on
+    void formatted_code_again;
+```
 
 #### Commit guidelines
 
-The purpose of these guidelines is to *make reviews easier* and make the VCS logs more valuable.
+The purpose of these guidelines is to *make reviews easier* and make the
+[VCS][vcs] logs more valuable.
 
 - Try to keep the first line under 70 characters.
-- Include further description, if necessary, after a blank line.
-    - Don't make it too verbose by including obvious things.
-    - But don't spare clarifications for anything that could be not so obvious.
-      Some commit messages are pages long, and that's fine if there's no better
-      place for those comments to live.
+- If necessary, Include further description after a blank line.
+    - Don't make the description too verbose by including obvious things, but
+      don't spare clarifications for anything that could be not so obvious.
+      Some commit messages are pages long, and that's fine if there's no
+      better place for those comments to live.
     - **Recommended:** Prefix logically-related commits with a consistent
-      identifier at the beginning of each commit message.
+      identifier in each commit message.
       [For example](https://github.com/neovim/neovim/commits?author=elmart),
       the following commits are related by task (*Introduce vim namespace*) and
       scope (*Contrib YCM*).
-      <br/> `Introduce vim namespace: Contrib YCM: Fix style issues.`
-      <br/> `Introduce vim namespace: Contrib YCM: Fix build dir calculation`
+      <br/> `Introduce nvim namespace: Contrib YCM: Fix style issues`
+      <br/> `Introduce nvim namespace: Contrib YCM: Fix build dir calculation`
         - Subtasks can be *activity-oriented* (doing different things on the same area)
           or *scope-oriented* (doing the same thing on different areas).
     - Granularity helps, but it's conceptual size that matters, not extent size.
-- Use the imperative voice: "Fix bug" rather than "Fixed bug" or "Fixes bug."
+- Use the [imperative][imperative] voice: "Fix bug" rather than "Fixed bug" or "Fixes bug."
+
+## Reviewing pull requests
+
+### Reviewer checklist
+
+Many items were [taken from here][code-review-checklist].
+
+#### Code
+
+- Is the Travis build passing?
+- Does the code conform to the [style guide][style]?
+- Does the code work? Does it perform its intended function, the logic is
+  correct etc.
+- Is the code easily understood?
+- Is there any redundant or duplicate code?
+- Are there any unused variables?
+- Is the code as modular as possible?
+- Can any global/static variables be replaced?
+- Can any function attributes be used?
+- Are variables named intuitively? functions?
```

Comment:
It's a fairly common idiom in the English language.  It's an alternate ending to the previous question.